examples/main.go

```diff
@@ -17,7 +18,9 @@ func main() {
 	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
 
 	bal, err := balancer.NewBalancer(&balancer.Config{
-		StrategyType: balancer.RoundRobin,
+		Strategy: func(nodes balancer.Nodes) (balancer.Strategy, error) {
+			return balancer.NewRoundRobinStrategy(nodes)
```

Comment:
food for thought: 

`balancer.NewRoundRobinStrategy` signature could be changed to:

    func NewRoundRobinStrategy(nodes Nodes) func(Nodes) (Strategy, error) {
	  return func(Nodes) (Strategy, error) {
		if len(nodes) == 0 {
			return nil, ErrNoNodesProvided
		}

		return &RoundRobinStrategy{
			nodes,
			0,
			&errgroup.Group{},
			sync.RWMutex{},
		}, nil
	 }
    }